### PR TITLE
ONLINE LED is white when downloading update

### PR DESCRIPTION
**Pull request by fvasquez**

### CODE REVIEW

File: src/gateway_status.erl [hunk 1]
```diff
@@ -99,6 +107,15 @@ handle_info({ebus_signal, _Path, Signal, Msg}, State=#state{lock_signal=Signal})
         {error, _Error} ->
             {noreply, State}
     end;
+handle_info({ebus_signal, _Path, Signal, Msg}, State=#state{download_signal=Signal}) ->
+    case ebus_message:args(Msg) of
+        {ok, [true]} ->
+            {noreply, update_led(?LED_ONLINE, downloading, State)};
```
[the remark this answers is not in the document]
Comment (fvasquez) [reply src/gateway_status.erl]: You are correct.  We wouldn't want an unexpected `online` signal to change the LED to green before the download has completed.

[the remark this answers is not in the document]
Comment (fvasquez) [reply src/gateway_status.erl]: Fixed by ignoring `online` and `ready` events when `downloading`.  Online status is restored after `downloading` by querying `connman:state()`.

File: src/gateway_status.erl [hunk 2]
```diff
@@ -82,10 +90,20 @@ handle_info({init_led, Led}, State=#state{}) ->
 handle_info({ebus_signal, _Path, Signal, Msg}, State=#state{online_signal=Signal}) ->
     case ebus_message:args(Msg) of
         {ok, ["State", "online"]} ->
-            {noreply, update_led(?LED_ONLINE, online, State)};
+            case is_downloading(State) of
```
Comment (madninja) [inline src/gateway_status.erl]: since all decisions on what to do with an LED are in update_led, maybe move this logic there so it stays co-located?

Comment (fvasquez) [reply src/gateway_status.erl]: The `downloading` logic has been pushed down into `upload_led`.

**Pull request merged**